module/os/freebsd/zfs/zfs_vfsops.c

```diff
@@ -1028,6 +1028,8 @@ zfsvfs_setup(zfsvfs_t *zfsvfs, boolean_t mounting)
 		return (error);
 
 	zfsvfs->z_log = zil_open(zfsvfs->z_os, zfs_get_data);
+	zil_init_wmsum(&zfsvfs->z_kstat.dk_zil_sums);
```

Comment:
I think the init should happen in dataset_kstat_create(), you only need to set the pointer in the zil struct when the zil is opened so that zil writes can update the sums.

Reply:
It might be worth codifying this requirement by passing the sums pointer through zfs_open instead. That way you can't forget to set it.

Reply:
As per our discussion on slack, moved it to ```dataset_kstats_create```